### ISSUE TITLE
aireplay-ng incorrectly reports "No such BSSID available."

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#324]
**Reported by darkAudax on 8 Sep 2007 23:00 UTC**

Situation:

3 APs have the same SSIDs and are on the same channel.  The signal and RXQ is quite good.  The wireless card has been set to the correct channel and confirmed.  When you run the following commands, you get the error "No such BSSID available."  If you try multiple times, it eventually works.

```
aireplay-ng -9 -e ABCDE -a bssid  wlan0
18:53:16  No such BSSID available.

aireplay-ng -1 6000 -e ABCDE -a bssid -h macaddr -o 1 -q 10  wlan0
18:54:41  Waiting for beacon frame (BSSID: bssid)
18:54:43  No such BSSID available.
```

I suspect that the logic is finding the correct SSID but because there are multiple beacons for the same SSID with different BSSIDs, it fails.  When you keep retrying, it eventually gets a beacon with the correct SSID/BSSID combination then works.

**Comment by aircrack-ng** [comment]
**Modified by misterx on 8 Sep 2007 23:00 UTC**

**Comment by aircrack-ng** [comment]
**Comment by hirte on 8 Sep 2007 23:00 UTC**

(In #705) Fixed "No such BSSID". (Closes: #324)
